### PR TITLE
[7.x] Lock support for cache file driver

### PR DESCRIPTION
**Pull request by rogervila**
Hello,

[Atomic locks](https://laravel.com/docs/7.x/cache#atomic-locks) are currently supported for `memcached`, `dynamodb`, and `redis` cache drivers.

This PR aims to implement this functionality for `file` cache driver, which is useful for environments that cannot use the other ones.

Thank you.

### CODE REVIEW

File: src/Illuminate/Cache/FileLock.php [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+     * Attempt to acquire the lock.
+     *
+     * @return bool
+     */
+    public function acquire()
+    {
+        $now = Carbon::now();
+        $content = $this->getLockFileContent();
+
+        if (is_null($content)) {
```
Comment (taylorotwell) [inline src/Illuminate/Cache/FileLock.php:85]: In between these two lines there is a race condition where another process could acquire the lock and then this process would overwrite application's lock. Therefore, this entire implementation isn't safe.

Comment (rogervila) [reply src/Illuminate/Cache/FileLock.php:85]: You are right. Can you mark it as a draft pull request? I will modify the file access to make sure that is locked during the whole process, and add tests that assure it.

Comment (driesvints) [reply src/Illuminate/Cache/FileLock.php:85]: @rogervila feel free to send in a new pr

Comment (rogervila) [reply src/Illuminate/Cache/FileLock.php:85]: Thank you @driesvints :)

**Pull request closed without merging**